nat-lab/tests/utils/testing.py

```diff
@@ -10,8 +12,48 @@ def unpack_optional(opt: Optional[T]) -> T:
     return opt
 
 
-def test_name_safe_for_file_name():
+def get_current_test_case_and_parameters() -> Tuple[Optional[str], Optional[str]]:
+    """
+    Returns a tuple of strings representing the case name and parameters of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+    """
     test_name = os.environ.get("PYTEST_CURRENT_TEST")
-    if test_name is not None:
-        return "".join([x if x.isalnum() else "_" for x in test_name.split(" ")[0]])
-    return test_name
+    if test_name:
+        test_parts = test_name.split("::", 1)[-1].split(" ")[0].split("[", 1)
+        if len(test_parts) > 1:
+            return (
+                format_path_string(test_parts[0]),
+                format_path_string(test_parts[1]),
+            )
+        return (format_path_string(test_parts[0]), None)
```

Comment:
Last extra nit, feel free to ignore it. I would return either None when PYTEST_CURRENT_TEST is empty for some reason. If it's not I would return tuple of strings, not tuple of optionals. In case parameters are not present it should be an empty string. Then in `get_current_test_log_path` you wouldn't have to test if the `test_parameters` is None or not. It would be empty string so you can always concatenate it :).

Reply:
Yeah I thought about it, and agree, but then instead of `test_name, test_parameters = get_current_test_case_and_parameters()` directly, we would have to unpack the optional tuple first.

If we don't check `test_parameters` at all we will end up with a trailing "_", unless we prepend it in `get_current_test_case_and_parameters()` which I also don't prefer.